### ISSUE TITLE
resize before upload

### ISSUE THREAD

**Issue by elwadi** [balderdashy/skipper#91]
hi, how to resize pictures before upload using gm ?

**Comment by tjwebb** [comment]
You'd need to do that after upload, and before the file is persisted into the database. You can implement your gm logic in a sails controller.

**Comment by elwadi** [comment]
can you give me an example ?

**Comment by tjwebb** [comment]
I don't know how to give you an example without writing your code for you. Can you be more specific about what you're having problems with? Make sure you've read and understand the docs on sails controllers and services.

**Comment by elwadi** [comment]
where i should add gm functions ?
```javascript
req.file('properties').upload(function (err, uploadedFiles){
  if (err) return res.send(500, err);
  else{
       /* add informations to data base ....*/
      //Property.create({.........});
      // files saved  with origine sizes

   }
  });
```

**Comment by plato-cambrian** [comment]
Hi, I'm streaming my uploads directly to s3 using skipper-s3, but now I want to make thumbnails. is there a way to stream the upload into `gm` first and then to the skipper receiver?

gm's streaming docs: https://github.com/aheckmann/gm#streams

**Comment by sailsbot** [comment]
Thanks for posting, @elwadi.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by toovy** [comment]
@mikermcneil as main contributor maybe you could give some clues about how one could resize using e.g. graphicsmagick or imagesmagick. I've tried to pipe the upstream from req.file('image') into a graphicsmagic or imagesmagick pipe (via transform stream), but after that important upstream metadata seems to be missing (e.g. the fd property), so piping to the receiver of skipper-disk or skipper-s3 fails. I've oriented at https://github.com/bredikhin/sailsjs-skipper-thumbnail-example, but didn't get it to work. It would be great if you could decouple resizing from the adapter as e.g. in our case some installations will write to the disk and some to s3. Any idea how to accomplish it?

**Comment by OwaysisApp** [comment]
+1

**Comment by ShepelievD** [comment]
UP 👍 
any ideas how can I use it with sharp.js or gm.js?

**Comment by MuhammadZaidSE** [comment]
+1
